back/api/v1/authentication.go

```diff
@@ -0,0 +1,168 @@
+package v1
+
+import (
+	"4ctf/models"
+	"4ctf/utils"
+	"4ctf/views"
+	"database/sql"
+	"errors"
+	"time"
+
+	"github.com/savsgio/atreugo/v11"
+	"github.com/volatiletech/sqlboiler/v4/boil"
+)
+
+func setupAuthRoutes(api *Api, router *atreugo.Router) {
+	group := router.NewGroupPath("/auth")
+
+	group.POST("/login", WithDefaults(api, WithBody(login)))
+	group.POST("/register", WithDefaults(api, WithBody(register)))
+	group.GET("/logout", WithDefaults(api, logout))
+	group.GET("/me", WithDefaults(api, WithValidSession(profile)))
+}
+
+type LoginRequest struct {
+	Username string `json:"username" validate:"min=2,max=40,regexp=^[a-zA-Z]*$"`
+	Password string `json:"password" validate:"max=40,password"`
```

Comment:
Only 40 ? Can not go to 50 or more ? Why 40 ?

Reply:
Arbitrary, we can limit to 128 or 256 characters to avoid DOS and not limit users

Reply:
It is recommand to set the higher possible limit so 256 seems good to me :)